### DISCUSSION TITLE
Why wallet list is different when we use @walletconnect/ethereum-provider

### DISCUSSION THREAD

**Discussion by shaman236** [WalletConnect/walletconnect-monorepo#2489] [Developer Support] [answered]
Hi,

I am integrating the WalletConnectV2 now.
I have integrated in 2 ways 
- using @web3modal/ethereum @web3modal/html @wagmi/core viem
- using @walletconnect/ethereum-provider @web3modal/standalone

But these 2 Web3modals are showing the different wallet list.
For exam, MetamaskMobile is showing for @wagmi/core but not showing for ethereum-provider

Is there anyway to show MetamaskMobile and other wallets on ethereum-provider WalletConnect modal?

**Comment by xzilja** [comment, marked as answer]
There is a short description for each in the docs:
https://docs.walletconnect.com/2.0/web/web3modal/html/wagmi/installation
https://docs.walletconnect.com/2.0/web/web3modal/html/ethereum-provider/installation

`viem / wagmi` allows for extra functionality like chain switching, extension wallets, account view and it supports walletconnect protocol versions v1 and v2.

`@walletconnect/ethereum-provider` is a much simpler implementation, it only allows for connection with walletconnect wallets and it only supports v2. You don't see metamask in here because they have not yet released version of wallet that supports v2 (it's coming soon).
